lang/java/avro/src/test/java/org/apache/avro/generic/TestGenericData.java

```diff
@@ -552,4 +552,28 @@ public void testToStringRecursive() throws IOException {
     }
   }
 
+  @Test
+  public void testToString() throws IOException {
+    List<Field> fields = new ArrayList<Field>();
+    fields.add(new Field("nullstring1", Schema.create(Type.STRING), null, null));
+    fields.add(new Field("nullstring2", Schema.create(Type.STRING), null, null));
+    fields.add(new Field("string1", Schema.create(Type.STRING), null, null));
+    fields.add(new Field("string2", Schema.create(Type.STRING), null, null));
+    fields.add(new Field("int1", Schema.create(Type.INT), null, null));
+    fields.add(new Field("int2", Schema.create(Type.INT), null, null));
+    Schema schema = Schema.createRecord("Foo", "test", "mytest", false);
+    schema.setFields(fields);
+
+    Record testRecord = new Record(schema);
+
+    testRecord.put("nullstring1", null);
+    testRecord.put("nullstring2", null);
+    testRecord.put("string1", "Hello World");
```

Comment:
Minor: The compiler is going to intern the "Hello World" string so you get the same object, but it would be more clear if these tests used the same reference instead of relying on that. Like this:

```java
String s = "Hello World";
testRecord.put("string1", s);
testRecord.put("string2", s);
```